src/options.cpp

```diff
@@ -99,6 +99,10 @@ class arg_parser {
       this->is_ignoring_options_ = true;
       this->option_ = std::nullopt;
     } else if (this->current_arg()[0] == '-') {
+      if (std::strcmp(this->current_arg(), "-") == 0) {
+        this->option_ = std::nullopt;
+        return;
+      }
```

Comment:
I think checking for `-` like on line 94 would look better:
```suggestion
    } else if (std::strcmp(this->current_arg(), "-") == 0) {
      this->option_ = std::nullopt;
    } else if (this->current_arg()[0] == '-') {
```